File: python/leetcode/217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hash = {}
        for num in nums:
            if num in hash:
                return True
            else:
                hash[num] = 1
        return False


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hash = {}

        for i in range(len(nums)):
            if nums[i] not in hash:
                hash[nums[i]] = 1
            else:
                hash[nums[i]] += 1
        return any([True for key, value in hash.items() if value >= 2])


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums.sort()
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1]:
                return True
        return False

File: c#/leetcode/217-contains-duplicate.cs

public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        
        foreach(int num in nums) {
            if (dict.ContainsKey(num)) {
                return true;
            }
            else {
                dict.Add(num ,1);
            }
        }
        return false;
    }
}